src/AerialShip/SamlSPBundle/State/SSO/SSOState.php

```diff
@@ -20,83 +20,128 @@ class SSOState implements \Serializable
     /** @var string */
     protected $nameIDFormat;
 
-
-
-
+    /** @var \DateTime */
+    protected $createdOn;
 
     /**
      * @param string $providerID
+     *
+     * @return SSOState
      */
-    public function setProviderID($providerID) {
+    public function setProviderID($providerID)
+    {
         $this->providerID = $providerID;
+
+        return $this;
```

Comment:
Is that your personal flavor, or more official symfony/psr recommendation? I got used to them, so it would be good to have a rational not to use them.